flytekit/core/condition.py

```diff
@@ -176,8 +176,8 @@ def end_branch(self) -> Union[Condition, Promise]:
         """
         ctx = FlyteContextManager.current_context()
         # Let us mark the execution state as complete
-        ctx.execution_state.branch_complete()
-        if self._last_case:
+        ctx.execution_state.branch_complete()  # type: ignore
+        if self._last_case and self._selected_case:
```

Comment:
was this a bug?!?

Reply:
`self._selected_case` could be `None`. We have to check it, otherwise mypy will raise an error.